From 477348e94b35a61251818c81551d67fc3477edeb Mon Sep 17 00:00:00 2001
From: Wayne Zou <b36644@freescale.com>
Date: Mon, 7 May 2012 16:57:30 +0800
Subject: ENGR00181670 V4L2: Fix bug:when ubuntu start up, print some fb error
 messages

If only called mxc_vout_open/mxc_vout_release, fb_smem_len are uninitialized.
When ubuntu start up, print some error messages:

mxc_sdc_fb mxc_sdc_fb.1: Unable to allocate framebuffer memory
mxc_v4l2_output mxc_v4l2_output.0: ERR: fb_set_var.
mxc_sdc_fb mxc_sdc_fb.0: Unable to allocate framebuffer memory
mxc_v4l2_output mxc_v4l2_output.0: ERR: fb_set_var.

Signed-off-by: Wayne Zou <b36644@freescale.com>
---
 drivers/media/video/mxc/output/mxc_vout.c |   22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/drivers/media/video/mxc/output/mxc_vout.c b/drivers/media/video/mxc/output/mxc_vout.c
index aebe57f..e3e9483 100644
--- a/drivers/media/video/mxc/output/mxc_vout.c
+++ b/drivers/media/video/mxc/output/mxc_vout.c
@@ -1675,16 +1675,18 @@ static void release_disp_output(struct mxc_vout_output *vout)
 	set_window_position(vout, &pos);
 
 	if (get_ipu_channel(fbi) == MEM_BG_SYNC) {
-		console_lock();
-		fbi->fix.smem_start = vout->fb_smem_start;
-		fbi->fix.smem_len = vout->fb_smem_len;
-		vout->fb_var.activate |= FB_ACTIVATE_FORCE;
-		fbi->flags |= FBINFO_MISC_USEREVENT;
-		ret = fb_set_var(fbi, &vout->fb_var);
-		fbi->flags &= ~FBINFO_MISC_USEREVENT;
-		console_unlock();
-		if (ret < 0)
-			v4l2_err(vout->vfd->v4l2_dev, "ERR: fb_set_var.\n");
+		if ((vout->fb_smem_len != 0) && (vout->fb_smem_start != 0)) {
+			console_lock();
+			fbi->fix.smem_start = vout->fb_smem_start;
+			fbi->fix.smem_len = vout->fb_smem_len;
+			vout->fb_var.activate |= FB_ACTIVATE_FORCE;
+			fbi->flags |= FBINFO_MISC_USEREVENT;
+			ret = fb_set_var(fbi, &vout->fb_var);
+			fbi->flags &= ~FBINFO_MISC_USEREVENT;
+			console_unlock();
+			if (ret < 0)
+				v4l2_err(vout->vfd->v4l2_dev, "ERR: fb_set_var.\n");
+		}
 		console_lock();
 		fbi->flags |= FBINFO_MISC_USEREVENT;
 		fb_blank(fbi, FB_BLANK_UNBLANK);
-- 
1.7.9.5

